cap/modules/deposit/api.py

```diff
@@ -115,3 +119,28 @@ def clear(self, *args, **kwargs):
     #             raise  # TODO raise REST exception ...
 
     #     return super(CAPDeposit, self).publish(*args, **kwargs)
+
+    @index
+    def clone(self, pid=None, id_=None):
+        """Clone a deposit
+
+        Adds snapshot of the files when deposit is cloned
+        """
+        data = dict(self.dumps())
+        if not data['_deposit']['status'] == 'draft':
+            data['_deposit']['status'] == 'draft'
+        id_ = id_ or uuid.uuid4()
+        deposit_pid = cap_deposit_minter(id_, data)
```

Comment:
The problem is that the default ```create``` checks this https://github.com/inveniosoftware/invenio-deposit/blob/master/invenio_deposit/api.py#L233 and it will not create a new id for the cloned deposit. Thats why I replicate the procedure.